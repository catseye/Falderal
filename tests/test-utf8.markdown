    -> encoding: UTF-8

Falderal Test: UTF-8
--------------------

This is an example Falderal document which contains Unicode
characters, encoded in UTF-8 (this is the assumed encoding
of all Falderal documents which go beyond mere ASCII.)

    -> Functionality "Cat" is implemented by
    -> shell command "python cat.py"

    -> Tests for functionality "Cat"

Cat cats.

    | n ← ★
    = n ← ★

Cat dogs, too. (Intentional fail.)

    | n ← ★
    = m ← ★

    -> Functionality "Cat (file)" is implemented by
    -> shell command "python cat.py -f %(test-body-file) -o %(output-file)"

    -> Tests for functionality "Cat (file)"

Cat (file) cats.

    | n ← ★
    = n ← ★

Cat (file) dogs, too. (Intentional fail.)

    | n ← ★
    = m ← ★
