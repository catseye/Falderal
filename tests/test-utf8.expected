FAILED  : 

Cat dogs, too. (Intentional fail.)

Location: test-utf8.markdown, line 22
Function: Cat
Impl    : shell command "python cat.py"
Body    : n ← ★
Expected: output:
m ← ★
Actual  : output:
n ← ★

FAILED  : 

Cat (file) dogs, too. (Intentional fail.)

Location: test-utf8.markdown, line 37
Function: Cat (file)
Impl    : shell command "python cat.py -f %(test-body-file) -o %(output-file)"
Body    : n ← ★
Expected: output:
m ← ★
Actual  : output:
n ← ★

FAILED  : 

Cat (input) dogs with its input, too. (Intentional fail.)

Location: test-utf8.markdown, line 53
Function: Cat (input)
Impl    : shell command "python cat.py -f %(test-input-file) -o %(output-file)"
Body    : m ← ★
Expected: output:
m ← ★
Actual  : output:
n ← ♡

--------------------------------
Total test runs: 6, failures: 3
--------------------------------
